### PR TITLE
Add support for ansible-galaxy

### LINKED ISSUE #3301: Install dependencies for ansible-local with a ansible-galaxy file
It would be nice if you could just say:

```yml
{
  "type": "ansible-local",
  "playbook_file": "local.yml",
  "galaxy": "requirements.yml"
}
```

And then it would install the specified roles in that file.

Currently I am trying to make this work by doing:

```yml
{
  "type": "ansible-local",
  "command": "ansible-galaxy install -v -r requirements.yml -p roles/ && ansible-playbook",
  "playbook_file": "local.yml"
}
```

Like @rvangundy mentioned in #1000

### PR DESCRIPTION
**Pull request by JoakimLofgren**
This adds "galaxy_file" to the options.

Closes #3301

### CODE REVIEW

**Comment by JoakimLofgren** [comment]
I tried looking at the [provisioner tests](https://github.com/mitchellh/packer/blob/master/provisioner/ansible-local/provisioner_test.go) but they seem to only test configuration and error paths..

Should I try to write some anyways ?

**Comment by JoakimLofgren** [comment]
Tests seem to fail on retrieving dependencies?

**Comment by JoakimLofgren** [comment]
Ah, I managed to get gox working now.
Will try it manually with my use case at least.

**Comment by JoakimLofgren** [comment]
So I got it to run by building the binaries myself.

Although took me a while to figure out that I needed to install git on my amazon instance before trying to run ansible-galaxy. :)

File: provisioner/ansible-local/provisioner.go [hunk 1]
```diff
@@ -242,6 +268,26 @@ func (p *Provisioner) Cancel() {
 	os.Exit(0)
 }
 
+func (p *Provisioner) executeGalaxy(ui packer.Ui, comm packer.Communicator) error {
+	rolesDir := filepath.ToSlash(filepath.Join(p.config.StagingDir, "roles"))
+
+	// ansible-galaxy install -v -r requirements.yml -p roles/
+	command := fmt.Sprintf("cd %s && %s install -r %s -p %s",
+		p.config.StagingDir, p.config.GalaxyCommand, p.config.GalaxyFile, rolesDir)
```
Comment (JoakimLofgren) [inline provisioner/ansible-local/provisioner.go]: Galaxy file needs to be `filepath.Base`'d as here as well...

**Comment by rickard-von-essen** [comment]
Works nicely, thanks! Could you add docs and I'll merge it.

**Comment by JoakimLofgren** [comment]
@rickard-von-essen Alright, where?

**Comment by JoakimLofgren** [comment]
https://github.com/mitchellh/packer/blob/master/website/source/docs/provisioners/ansible.html.md I'm guessing there?

**Comment by rickard-von-essen** [comment]
ansible-local.html

**Comment by JoakimLofgren** [comment]
@rickard-von-essen wdyt of the doc I added?

**Comment by rickard-von-essen** [comment]
Looks good!

**Pull request merged**